client/post-editor/editor-sidebar/header.jsx

```diff
@@ -54,9 +54,8 @@ function EditorSidebarHeader( { translate, type, siteId, showDrafts, toggleDraft
 				</Button>
 			) }
 			{ type === 'post' && siteId && (
-				<PostCountsData siteId={ siteId } status="draft">
-					<DraftsButton onClick={ toggleDrafts } />
-				</PostCountsData>
+				<QueryPostCounts siteId={ siteId } type="post" /> ) && (
```

Comment:
Funny I actually had it in the `<DraftsButton />` at one point while working on the branch.  Thanks for confirming that hunch.